knotx-core/knotx-fragment-assembler/src/main/java/com/cognifide/knotx/knot/assembler/FragmentAssemblerVerticle.java

```diff
@@ -0,0 +1,124 @@
+import java.net.URISyntaxException;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import rx.Observable;
+
+public class FragmentAssemblerVerticle extends AbstractVerticle {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FragmentAssemblerVerticle.class);
+
+  private static final String SNIPPET_IDENTIFIER_NAME = "data-knot-types";
+
+  private static final String ANY_SNIPPET_PATTERN =
+      "(?is).*<script\\s+" + SNIPPET_IDENTIFIER_NAME + ".*";
+
+  private FragmentAssemblerConfiguration configuration;
+
+  @Override
+  public void init(Vertx vertx, Context context) {
+    super.init(vertx, context);
+    this.configuration = new FragmentAssemblerConfiguration(config());
+  }
+
+  @Override
+  public void start() throws IOException, URISyntaxException {
+    LOGGER.debug("Starting <{}>", this.getClass().getName());
+    EventBus eventBus = vertx.eventBus();
+
+    eventBus.<KnotContext>consumer(configuration.getAddress()).handler(
+        message -> Observable.just(message)
+            .doOnNext(this::traceMessage)
+            .subscribe(
+                response -> {
+                  KnotContext context = response.body();
+                  response.reply(createSuccessResponse(context, getTemplateContent(context)));
+                },
+                error -> {
+                  LOGGER.error("Exception happened during HTML splitting.", error);
+                  message.reply(processError(message.body(), error));
+                }
+            )
+    );
+  }
+
+  private String getTemplateContent(KnotContext context) {
+    return context.fragments().map(fragments -> fragments.stream()
+        .map(Fragment::getContent)
+        .filter(StringUtils::isNotBlank)
+        .map(value -> {
+          if (value.matches(ANY_SNIPPET_PATTERN)) {
+            Document document = Jsoup.parseBodyFragment(value);
+            Element scriptTag = document.body().child(0);
+            return scriptTag.unwrap().toString();
+          } else {
+            return value;
+          }
+        })
+        .collect(Collectors.joining())).orElse(StringUtils.EMPTY);
+  }
+
+  private KnotContext processError(KnotContext knotContext, Throwable error) {
+    LOGGER.error("Error happened during Template processing", error);
+    ClientResponse errorResponse = new ClientResponse()
+        .setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+
+    return new KnotContext()
+        .setClientRequest(knotContext.clientRequest())
+        .setClientResponse(errorResponse);
+  }
+
+  private KnotContext createSuccessResponse(KnotContext inputContext, String renderedContent) {
+    ClientResponse clientResponse = inputContext.clientResponse();
+    MultiMap headers = clientResponse.headers();
+    headers.set(HttpHeaders.CONTENT_LENGTH.toString(), Integer.toString(renderedContent.length()));
```

Comment:
For clarity of code, first three lines I would make a method:
```java
private ClientResponse responseWithLength(KnotContext inputContext, String content);
```
To clearly see what's being done here. Client Reponse is retrieved and length is being set in headers